lxd/container_lxc.go

```diff
@@ -2319,6 +2326,66 @@ func (c *containerLXC) startCommon() (string, error) {
 				}
 			}
 
+			// Creating loopback device inside container namespace for IPVLAN network
+			if m["type"] == "nic" && m["nictype"] == "ipvlan" {
```

Comment:
I don't expect any of those sysctl calls to be acceptable.

It's fine to tweak sysctls on interfaces that we ourselves create and own (bridges, container devices), but it certainly isn't fine to just start messing with global system settings because one container requested a particular interface type.

Some of the options below would just plain break connectivity on my system was I to even just run our testsuite with this change applied...

Reply:
(well, assuming we had a test for ipvlan)

Reply:
But all of those syscals are applied to parent IPVLAN interface, except ip_forward.
We already own exclusively parent interface if we use IPVLAN or MACVTAP or bridge type connectivity, we can not reuse different types of networking for same parent interface, kernel will issue an error if one would try to use  MACVTAP bridge with parent interface currently used for IPVLAN.

What changes do you propose?

Reply:
Document the needed host configuration in a document in doc/ and if you want, add a check for the sysctl values, but don't modify a system interface that wasn't created by LXD.

Unsetting autoconf/accept_ra would make my machine loose IPv6 connectivity, turning the global ip_forward flag to 1 (instead of per-interface forwarding) means that two completely unrelated network interfaces may now be able to forward traffic I don't want, ...

Reply:
should be fixed by:
https://github.com/s3rj1k/lxd/commit/d8cb08b2f957c5dd431e2f6953a51f8b0f3afeb5
https://github.com/s3rj1k/lxd/commit/9927faf59df53c5e0e853ffa28b7d5e5cbb59ab8

Reply:
resolved in https://github.com/s3rj1k/lxd/commit/667be539f1ea3aabd92d5a826aa911783401a272